### ISSUE TITLE
Feature Map Visualization

### ISSUE THREAD

**Issue by dalaoplan** [PhysiologicAILab/FactorizePhys#1]
Very interesting work, I would like to ask if the code for visualizing the feature maps in the paper is available?

**Comment by jnj256** [comment]
Thank you, @dalaoplan for finding our work interesting,

The code for the visualization of feature maps can be found here -> 

https://github.com/PhysiologicAILab/FactorizePhys/blob/8cd115145eb0a3c17941a4286a224a49d2ab687c/neural_methods/model/FactorizePhys/test_FactorizePhys.py#L162 

And

https://github.com/PhysiologicAILab/FactorizePhys/blob/8cd115145eb0a3c17941a4286a224a49d2ab687c/neural_methods/model/FactorizePhys/test_FactorizePhysBig.py#L168

If you want higher resolution of feature maps for FactorizePhys.py (that takes 72x72 spatial resolution video frames), you can move following three lines of code from rPPG_FeatureExtractor to the begining of self.conv_block of BVP_Head:

            ConvBlock3D(nf[1], nf[1], [3, 3, 3], [1, 1, 1], [1, 0, 0]), #B, nf[1], 160, 31, 31
            ConvBlock3D(nf[1], nf[2], [3, 3, 3], [1, 2, 2], [1, 0, 0]), #B, nf[2], 160, 15, 15
            ConvBlock3D(nf[2], nf[2], [3, 3, 3], [1, 1, 1], [1, 0, 0]), #B, nf[2], 160, 13, 13
            nn.Dropout3d(p=dropout_rate),

**Comment by dalaoplan** [comment]
@jnj256 I have a question as to what I should do if I want to superimpose the attention map from the paper on the input image as shown below.
<img width="461" alt="d811e3344c6d82013176cc9eaa1cb90" src="https://github.com/user-attachments/assets/a2624d3e-38b3-4b90-8150-a2eca01bbdff" />

**Comment by jnj256** [comment]
This is quite easily possible with matplotlib. For example, in test_FactorizePhys.py, you can add transparency by setting alpha value in the imshow something like following:

        ch = 0
        ax[0, 1].imshow(self.np_data[enc_frames//2, ...].astype(np.uint8))
        ax[0, 1].imshow(corr_matrix[0, ch, :, :], cmap=cmap, vmin=0, vmax=1, alpha=0.5)
        ax[0, 1].axis('off')

You can do this for each channel and may try few different alpha levels and see whats more convincing.

**Comment by dalaoplan** [comment]
> This is quite easily possible with matplotlib. For example, in test_FactorizePhys.py, you can add transparency by setting alpha value in the imshow something like following:
> 
> ```
>     ch = 0
>     ax[0, 1].imshow(self.np_data[enc_frames//2, ...].astype(np.uint8))
>     ax[0, 1].imshow(corr_matrix[0, ch, :, :], cmap=cmap, vmin=0, vmax=1, alpha=0.5)
>     ax[0, 1].axis('off')
> ```
> 
> You can do this for each channel and may try few different alpha levels and see whats more convincing.

It worked. Thank you so much.
